src/azure-cli/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -540,6 +547,115 @@ def cli_cosmosdb_sql_trigger_update(client,
                                             sql_trigger_create_update_resource)
 
 
+def cli_cosmosdb_sql_role_definition_create(client,
+                                            resource_group_name,
+                                            account_name,
+                                            role_definition_body):
+    '''Creates an Azure Cosmos DB SQL Role Definition '''
+    print(555)
+    role_definition_create_resource = SqlRoleDefinitionCreateUpdateParameters(
+        role_name=role_definition_body['RoleName'],
+        type=role_definition_body['Type'],
+        assignable_scopes=role_definition_body['AssignableScopes'],
+        permissions=role_definition_body['Permissions'])
+    print(561)
+    returnval = client.create_update_sql_role_definition(role_definition_body['Id'], resource_group_name, account_name, role_definition_create_resource)
+    print(563)
+    return returnval
+
+
+def cli_cosmosdb_sql_role_definition_update(client,
+                                            resource_group_name,
+                                            account_name,
+                                            role_definition_body):
+    '''Update an existing Azure Cosmos DB Sql Role Definition'''
+    logger.debug('reading SQL role definition')
+    role_definition = client.get_sql_role_definition(role_definition_body['Id'], resource_group_name, account_name)
+
+    if role_definition_body['RoleName'] is not None:
+        role_definition.role_name = role_definition_body['RoleName']
+
+    if role_definition_body['AssignableScopes'] is not None:
+        role_definition.assignable_scopes = role_definition_body['AssignableScopes']
+
+    if role_definition_body['Permissions'] is not None:
+        role_definition.permissions = role_definition_body['Permissions']
+
+    role_definition_update_resource = SqlRoleDefinitionCreateUpdateParameters(
+        role_name=role_definition.role_name,
+        type=role_definition_body['Type'],
+        assignable_scopes=role_definition.assignable_scopes,
+        permissions=role_definition.permissions)
+
+    return client.create_update_sql_role_definition(role_definition_body['Id'], resource_group_name, account_name, role_definition_update_resource)
+
+
+def cli_cosmosdb_sql_role_definition_exists(client,
+                                            resource_group_name,
+                                            account_name,
+                                            role_definition_id):
+    """Checks if an Azure Cosmos DB Sql Role Definition exists"""
+    try:
+        client.get_sql_role_definition(role_definition_id, resource_group_name, account_name)
+    except CloudError as ex:
+        return _handle_exists_exception(ex)
```

Comment:
Can `CloudError` here contain other exceptions than exists exception?

Reply:
I changed this to DefaultErrorResponseException, but if you check _handle_exists_exception() if will check for a 404.